spring-boot-project/spring-boot/src/main/java/org/springframework/boot/DigitalUnit.java

```diff
@@ -0,0 +1,498 @@
+/*
+ * Copyright 2012-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot;
+
+import org.springframework.lang.NonNull;
+import org.springframework.lang.Nullable;
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+
+/**
+ * A {@code DigitalUnit} represents digital amount at a given unit of granularity and
+ * provides utility methods to convert across units.
+ *
+ * @author Dmytro Nosan
+ * @since 2.1.0
+ */
+public enum DigitalUnit {
```

Comment:
This should be moved to a nested `Unit` of `DataSize`. It doesn't have to be that complicated either. We don't need to expose all the unit methods there

I had crafted something like this locally

```
public enum Unit {

        /**
         * KiloByte.
         */
        KILOBYTES((size) -> size * BYTES_PER_KB),

        /**
         * MegaByte.
         */
        MEGABYTE((size) -> size * BYTES_PER_MB);

        private final Function<Long, Long> converter;

        Unit(Function<Long, Long> converter) {
            this.converter = converter;
        }

        public long toBytes(long size) {
            return this.converter.apply(size);
        }

    }
```